Mathlib/LinearAlgebra/Matrix/SchurTriangulation.lean

```diff
@@ -0,0 +1,300 @@
+/-
+Copyright (c) 2025 Gordon Hsu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gordon Hsu
+-/
+import Mathlib.Analysis.InnerProductSpace.Adjoint
+import Mathlib.LinearAlgebra.Eigenspace.Triangularizable
+/-! # Schur triangulation
+
+Schur triangulation is more commonly known as Schur decomposition or Schur triangularization, but
+"triangulation" makes the API more readable. It states that a square matrix over an algebraically
+closed field, e.g., `ℂ`, is unitarily similar to an upper triangular matrix.
+
+## Main definitions
+
+- `Matrix.schur_triangulation` : a matrix `A : Matrix n n 𝕜` with `𝕜` being algebraically closed can
+be decomposed as `A = U * T * star U` where `U` is unitary and `T` is upper triangular.
+- `Matrix.schurTriangulationUnitary` : the unitary matrix `U` as previously stated.
+- `Matrix.schurTriangulation` : the upper triangular matrix `T` as previously stated.
+- Some auxilary definitions are not meant to be used directly, but
+`LinearMap.SchurTriangulationAux.of` contains the main algorithm for the triangulation procedure.
+
+-/
+
+section
+
+variable {m n : ℕ}
+
+/-- `subNat' i h` subtracts `m` from `i`. This is an alternative form of `Fin.subNat`. -/
```

Comment:
Can you elaborate on how it differs?

Reply:
`Fin.subNat` takes a `Fin (n + m)` while `Fin.subNat'`
1. takes a `Fin (m + n)`,
2. makes `m` implicit, and
3. accepts `¬ i < m` instead of `m ≤ i`.

I intend `Fin.subNat'` to be a placeholder for `Sigma.snd : Fin n` when `Σ b, cond b (Fin m) (Fin n)` has `Sigma.fst = false`. The placeholder just so happened to have a type similar to `Fin.subNat`, so I named it `Fin.subNat'`.